### PR TITLE
Basic logic for moves and attacks

### PR DESCRIPTION
**Pull request by Edwardb11**
nextTurn which is for the next turn as the name implies ...
is selected to return what I am selecting
move as I said, this is the logic of the game, it is controlled if the piece is already queen as well as if it is occupied and depending on this it can remove (I mean eat) as for the click method as its name indicates it fills the element with another color selected

### CODE REVIEW

**Comment by jorgelinor** [comment]
Little description please?

**Comment by Edwardb11** [comment]
nextTurn which is for the next turn as the name implies ...
is selected to return what I am selecting
move as I said, this is the logic of the game, it is controlled if the piece is already queen as well as if it is occupied and depending on this it can remove (I mean eat) as for the click method as its name indicates it fills the element with another color selected

File: board.py
Comment (jorgelinor) [inline board.py]: I think you can abstract this "repeated" code to a helper function. This function will receive the only visible variable in all these lines: the first array. So you would simplify all of this and make more readable. This should be the result:
```
        if piece.isKing:
            max_valid_moves, max_valid_attacks = attacks_and_moves([-1, 1], [-2, 2])
        else:
            if self.turn == 0:
                max_valid_moves, max_valid_attacks = attacks_and_moves([-1], [-2])
            else:
                max_valid_moves, max_valid_attacks = attacks_and_moves([1], [2])
```
Reminder: the helper function should return `(max_valid_moves, max_valid_attacks)` to make this work correctly

**Comment by jorgelinor** [comment]
I asked for the description for this PR 😂 like "Basic logic for moves and attacks". Not all your PRs will be known as "Game", giving your reviewer a summary of what he will review eases the work for him and both save time since there's less to ask

**Review by jorgelinor — CHANGES_REQUESTED** [review]
In general it looks good, just a few points to improve:
- it is hard to read the `move` function, try creating functions for every repeated logic without breaking it
- use comments to document your code or clarify something that you could take more than 10 seconds to understand
- the `move` function is way too big, can you consider make it smaller? I mean, use for that specific action: to move. The part that evaluates if the tile can move or can attack and even the part of attack should have their own methods
- when you say "board.move()" don't you expect the board to move instead of tiles? 🤔 

Oh well, they were more points than I expected 😅 try improving your code according to them and let me know so I can check again

File: main.py
Comment (jorgelinor) [inline main.py:11]: Please make sure to use this property in line 22

**Review by jorgelinor — APPROVED** [review]
Very nice work! Just a little change to address before get it merge but in general it's amazing! When you have addressed the change according to the comment below please let me know to approve it again (only if you lose my ✅

File: main.py
Comment (Edwardb11) [reply main.py:11]: done

**Pull request merged**
